### ISSUE TITLE
BIMI support?

### ISSUE THREAD

**Issue by strarsis** [mailcow/mailcow-dockerized#3671]
**Is your feature request related to a problem? Please describe.**
The new BIMI feature allows to add a brand logo/indicator to supporting mail clients.

**Describe the solution you'd like**
A BIMI integration feature, probably involves some kind of repository for brand logos, DNS records and such.

**Additional context**
https://bimigroup.org/

**Comment by mkuron** [comment]
Seems like it‘s in a very early stage, with only Yahoo and Gmail supporting it on the receiving end and no clear specification.

On the sender side, Mailcow already supports it because it only requires a DNS record. Apparently in some situations you also need a Verified Mark Certificate, which I found little information on but it seems conceptually similar to an Extended Validation certificate. Mailcow can‘t help you with that.

On the receiving side, you need client support. File a feature request with SOGo if you want to see logos in our webmail or with Mozilla/Microsoft/Apple/... if you want to see them on your computer or smart phone.

**Comment by Adorfer** [comment]
is there any reference implementation?  Is there an RFC?

this website sound(!) like to cross/mix different aspects of mail like dmarc and MUA-branding, which sounds -for me- a bit confusing.

**Comment by dragoangel** [comment]
Doesn't understand how your one TXT DNS record and one svg on any https server related to mailcow. You can implement BIMI by yourself.

Also to note:
Currently BIMI isn't require Company Logo Certificate, but at same time now it simply doesn't work based on my experience. But in future you will need to buy this certificate for BIMI and add public part to DNS TXT record which will proof of that you are real company and your logo is really yours. Again: how mailcow related to this?

**Comment by strarsis** [comment]
Right, so I found that BIMI thing and thought that it maybe helps delivering email.
I am really interested in improving mailcow and so I thought that throwing "BIMI" in could help making other postmasters aware of this feature.

Though BIMI apparently only requires DNS/HTTP(S), mailcow could be still involved in this, namely in showing the right DNS records, mailcow also already offers some basic logo brainding of its UI. Also, mailcow offers HTTPS mail client (SoGo), so letting it also server the BIMI logo files, etc and offer some instructions to the postmaster would be a nice and fitting addition IMHO.

**Comment by Adorfer** [comment]
"improving mailcow" is a valid proposition
"but you can implement that in your local installation": perhaps not the focus of an "all in one" package. Since otherwise we are on the slippery slope towards _"nobody needs mailcow, you can install sogo, rpamd etc yourself on top of postfix and dovecot"_ 

the basic problem with BIMI i see (please prove me wrong)

1. lack of RFC or other  easy accesible technical document describing the standard
2. lack of common MUA using it (neither thunderbird, nor outlook, not k9, nor -you name it-) 
3. lack of server side reference implementations / setup-how-tos in the FOSS universe

in other words: I like the concept, but for me it's to early (esp. due to point 2) to deal with it.

**Comment by dragoangel** [comment]
Hi @strarsis BIMI apparently reqire VMC, but if you like to have How-to with DNS and HTTPS, here it is:
1. Login to mailcow server
2. Upload svg to Data/web folder inside mailcow directory.
3. Check that you can access this svg file via browser with link: https://mailcowhostname/logo.svg
4. Add TXT record to your domain with link to svg, but without VerifiedMarcCertificate: `v=BIMI1;l=https://mailcowhostname/logo.svg`.
5. Check that you have reject DMARC policy in place with DKIM & SPF.
6. Test that mail you send has successfully pass SPF & DKIM.
7. Test BIMI logo with online tool you already know. You will see nice logo from your svg.
8. Send real mail from your email on configured domain to Yahoo & Gmail and see that there `no logo` in place you want to see it.


And if we speak about real working BIMI setup, write email to DigiCert or Valimail and first of all buy VMC from them. They're will say something like: you need have own registered trademark and registered logo, only after you provide docs & approved ownership of domain process can continue. I actually think when you will get price of all of this + VMC you will stop to dig in this way, really. Their prices not open publicly, but I believe they will be cost a lot, much more then EV certificate cost now, specially due to hipe & monopoly. If you get this certificate, they will give you all needed instructions how to apply it and get this working with all support :). And if you will have enough money to pay for trademark, registered logo and VMC - question about hosting one single SVG will be not an issue for you (which actually even not an img - this is text, xml representation of vectors which can be hosted even on your gdrive).

**Comment by strarsis** [comment]
The VMC certificate is currently optional, which is nice.

The BIMI site offers a kind of generator/helper for setting up BIMI:
https://bimigroup.org/bimi-generator/

**Comment by dragoangel** [comment]
@strarsis did you get your logo in reality in gmail or Yahoo?) Of course - no. Optional or not - it not working without it. I have bimi more then half year ago, and? Not saw it ever works

**Comment by strarsis** [comment]
@dragoangel: Oh, it doesn't work yet (or never?)?

**Comment by dragoangel** [comment]
@strarsis please read my previous comment. I clearly said: it never works without VMC and I already gived you full how to setup manual as well to check it.

**Comment by Sysadminfromhell** [comment]
bump?
